erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -253,17 +253,25 @@ def get_directions(self, route, optimize):
 
 
 @frappe.whitelist()
-def get_delivery_window(customer=None):
-	delivery_window = frappe.db.get_value("Customer", customer, ["delivery_start_time", "delivery_end_time"], as_dict=1)
-	delivery_window.default_window = False
-
-	if delivery_window and (delivery_window.delivery_start_time or delivery_window.delivery_end_time):
+def get_delivery_window(doctype=None, docname=None, customer=None):
+	if doctype and docname:
+		delivery_window = frappe.db.get_value(doctype, docname , ["delivery_start_time", "delivery_end_time"], as_dict=1)
+		delivery_window.default_window = False
 		return delivery_window
+	elif customer:
+		delivery_window = frappe.db.get_value("Customer", customer, ["delivery_start_time", "delivery_end_time"], as_dict=1)
+		delivery_window.default_window = False
 
+		if delivery_window and (delivery_window.delivery_start_time or delivery_window.delivery_end_time):
+				return delivery_window
+		
 	default_window = frappe.db.get_value("Delivery Settings", "Delivery Settings", ["delivery_start_time", "delivery_end_time"], as_dict=1)
-	default_window.default_window = True
-	return default_window
-
+	if default_window.delivery_start_time and default_window.delivery_end_time:
+		default_window.default_window = True
+		return default_window
+	else:
+			frappe.throw(_("Default Delivery Window not available"))
```

Comment:
1. Remove the `if..else` condition checks.
2. Don't stop execution if nothing is found. Since this is called in multiple transactions, you'll be breaking flows, just return null values, that's fine, the delivery window is not mandatory.